packages/ledgerstate/address.go

```diff
@@ -0,0 +1,265 @@
+	marshalUtil := marshalutil.New(bytes)
+	if address, err = AddressFromMarshalUtil(marshalUtil); err != nil {
+		err = xerrors.Errorf("failed to parse Address: %w", err)
+	}
+	consumedBytes = marshalUtil.ReadOffset()
+
+	return
+}
+
+// AddressFromBase58EncodedString creates an Address from a base58 encoded string.
+func AddressFromBase58EncodedString(base58String string) (address Address, err error) {
+	bytes, err := base58.Decode(base58String)
+	if err != nil {
+		err = xerrors.Errorf("error while decoding base58 encoded Address (%v): %w", err, ErrBase58DecodeFailed)
+		return
+	}
+
+	if address, _, err = AddressFromBytes(bytes); err != nil {
+		err = xerrors.Errorf("failed to parse Address: %w", err)
+		return
+	}
+
+	return
+}
+
+// AddressFromMarshalUtil reads an Address from the bytes in the given MarshalUtil.
+func AddressFromMarshalUtil(marshalUtil *marshalutil.MarshalUtil) (address Address, err error) {
+	addressType, err := marshalUtil.ReadByte()
+	if err != nil {
+		err = xerrors.Errorf("failed to parse AddressType (%v): %w", err, ErrParseBytesFailed)
+		return
+	}
+	marshalUtil.ReadSeek(-1)
+
+	switch AddressType(addressType) {
+	case AddressTypeED25519:
+		return ED25519AddressFromMarshalUtil(marshalUtil)
+	case AddressTypeBLS:
+		return BLSAddressFromMarshalUtil(marshalUtil)
+	default:
+		err = xerrors.Errorf("unsupported address type (%X): %w", addressType, ErrParseBytesFailed)
+		return
+	}
+}
+
+// endregion ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// region ED25519Address ///////////////////////////////////////////////////////////////////////////////////////////////
+
+// ED25519Address represents an Address that is secured by the ED25519 signature scheme.
+type ED25519Address struct {
+	digest []byte
+}
+
+// NewED25519Address creates a new ED25519Address from then given public key.
+func NewED25519Address(publicKey ed25519.PublicKey) *ED25519Address {
+	digest := blake2b.Sum256(publicKey[:])
+
+	return &ED25519Address{
+		digest: digest[:],
+	}
+}
+
+// ED25519AddressFromBytes unmarshals an ED25519Address from a sequence of bytes.
+func ED25519AddressFromBytes(bytes []byte) (address *ED25519Address, consumedBytes int, err error) {
+	marshalUtil := marshalutil.New(bytes)
+	if address, err = ED25519AddressFromMarshalUtil(marshalUtil); err != nil {
+		err = xerrors.Errorf("failed to parse ED25519Address: %w", err)
+		return
+	}
+	consumedBytes = marshalUtil.ReadOffset()
+
+	return
+}
+
+// ED25519AddressFromMarshalUtil is a method that parses an ED25519Address from the given MarshalUtil.
+func ED25519AddressFromMarshalUtil(marshalUtil *marshalutil.MarshalUtil) (address *ED25519Address, err error) {
+	addressType, err := marshalUtil.ReadByte()
+	if err != nil {
+		err = xerrors.Errorf("failed to parse AddressType (%v): %w", err, ErrParseBytesFailed)
+		return
+	}
+	if AddressType(addressType) != AddressTypeBLS {
```

Comment:
Shouldn't this be `AddressTypeED25519`?
```suggestion
	if AddressType(addressType) != AddressTypeED25519{
```